### PR TITLE
Adds sample app

### PR DESCRIPTION
**Pull request by ecgreb**
Adds a sample app module to the repo with MParticle & Button Kit integration that can be used for testing and documentation.

### CODE REVIEW

**Comment by najmsheikh** [comment]
This would be a good opportunity for someone who has _not_ been working on mParticle related tasks to try and deploy, edit, and test the project.

File: sample/src/main/java/com/example/buttonkit/IntentActivity.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_intent)
+
+        actionBar?.setDisplayShowHomeEnabled(true)
+
+        val button = MParticle.getInstance().getKitInstance(
+                MParticle.ServiceProviders.BUTTON) as ButtonKit?
+
+        tv_attribution_token.text = button?.attributionToken
```
Comment (ecgreb) [inline sample/src/main/java/com/example/buttonkit/IntentActivity.kt:20]: Missing import `import kotlinx.android.synthetic.main.activity_intent.tv_attribution_token`

File: sample/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.buttonkit">
+
+    <application
+        android:name=".SampleApplication"
+        android:allowBackup="true"
```
Comment (ecgreb) [inline sample/src/main/AndroidManifest.xml]: Let's switch to `false` so uninstalling actually clears data 😂

File: sample/README.md [hunk 3]
```diff
@@ -34,4 +34,5 @@ If you recently issued new credentials, try re-installing the sample app as the
 Confirm the following:
 1. The proper package name (e.g. `com.example.buttonkit`) is shown in the mParticle Dashboard.
 2. There exists an input (the sample/test app) and an output (Button integration from mParticle marketplace) and that it is active.
-3. The Button integration is configured properly with a valid [Application ID](https://app.usebutton.com/account/login/?next=/settings/organization)
+3. The Button integration is configured properly with a valid [Application ID](https://app.usebutton.com/account/login/?next=/settings/organization).
```
Comment (ecgreb) [inline sample/README.md]: Can we specify _where_ the Button App id is configured (i.e. mParticle dashboard)

File: sample/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<resources>
+    <string name="app_name">ButtonKit Sample</string>
```
Comment (ecgreb) [inline sample/src/main/res/values/strings.xml]: Let's shorten sample app name to just "ButtonKit" so it doesn't truncate in the launcher app :)

File: sample/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+#### Receiving the following error: `Bad API request - is the correct API key and secret configured?`
+Confirm that mParticle SDK credentials are being properly retrieved in the `build.gradle` file as well as if the credentials themselves are correct.
+If you recently issued new credentials, try re-installing the sample app as the credentials may have been cached.
+
+#### ButtonKit is not initializing
+Confirm the following:
+1. The proper package name (e.g. `com.example.buttonkit`) is shown in the mParticle Dashboard.
+2. There exists an input (the sample/test app) and an output (Button integration from mParticle marketplace) and that it is active.
+3. The Button integration is configured properly with a valid [Application ID](https://app.usebutton.com/account/login/?next=/settings/organization).
+4. You are not behind a proxy (e.g. Charles Proxy).
```
Comment (ecgreb) [inline sample/README.md]: Thank you for adding this - it definitely got me on the first try!

**Comment by ecgreb** [comment]
LGTM

**Pull request merged**